invokeai/app/invocations/latent.py

```diff
@@ -912,6 +932,8 @@ def invoke(self, context: InvocationContext) -> LatentsOutput:
                 do_classifier_free_guidance=True,
             )
 
+            image_prompts = self.prep_ip_adapter_image_prompts(context=context, ip_adapter=self.ip_adapter)
+
```

Comment:
Can you add a comment to this call explaining:
- Why this has been split out from `prep_ip_adapter_data()`
- The importance of calling one inside the big model context manager and the other outside of it.
- The implications around increased likelihood of model switching.

Reply:
I added a comment addressing 1 and 2, but I'm not sure what you mean by model switching.

Reply:
Just that since we now lock->unlock->lock the IP-Adapter it could be unloaded / re-loaded. Whether this actually happens depends on a bunch of factors, of course.